Lesson1/T_Ilya/CPP_ADV_Lesson1/functions.cpp

```diff
@@ -0,0 +1,64 @@
+#include "functions.h"
+#include "task.h"
+
+namespace MFun{
+void print_vect(std::vector<int> a_vect){
+    for(MySpace::ull i{0}; i < a_vect.size(); i++){
+        std::cout << a_vect[i] << " , ";
+    }std::cout << std::endl;
+
+}
+int divide_first(int x){
+    while(x >= 10){
+        x /= 10;
+    }
+    return x;
+}
+int divide_last(int y){
+    return (y % 10);
+
+}
+void power(int c, int d, std::vector<int>& vect_c){
+    int r{1};
+    for (auto i{0}; i < d; ++i){
+        r *= c;
+        vect_c.emplace_back(r);
+    }
+}
+void print_vect_b(const std::vector<int>& vect_c){
+    for(const int& value: vect_c)
+    {
+        std::cout << value << " , ";
+    }
+    std::cout << std::endl;
+}
+void showlist_z(std::list <int> list_z){
```

Comment:
1. Naming - what is z? Why? Please write methods in generic way. So, for example, why print_vect_b and print_vect exist? They do the same things.
2. **Do not pass constructors by copy, if it is not needed**